server/hotrod/src/test/java/org/infinispan/server/hotrod/logging/HotRodAccessLoggingTest.java

```diff
@@ -0,0 +1,33 @@
+package org.infinispan.server.hotrod.logging;
+
+import static org.testng.AssertJUnit.assertTrue;
+
+import java.lang.reflect.Method;
+
+import org.apache.logging.log4j.Level;
+import org.infinispan.commons.test.skip.StringLogAppender;
+import org.infinispan.server.hotrod.test.HotRodSingleNodeTest;
+import org.testng.annotations.Test;
+
+/**
+ * @since 9.2
+ */
+@Test(groups = "functional", testName = "server.hotrod.logging.HotRodAccessLoggingTest")
+public class HotRodAccessLoggingTest extends HotRodSingleNodeTest {
+   StringLogAppender logAppender;
+
+   @Override
+   protected void setup() throws Exception {
+      logAppender = new StringLogAppender("org.infinispan.HOTROD_ACCESS_LOG", Level.TRACE, t -> t.getName().startsWith("HotRod-HotRodAccessLoggingTest-ServerWorker"));
+      logAppender.install();
+      super.setup();
+   }
+
+   public void testRestAccessLog(Method m) {
+      client().put("key", "value");
+
+      String logline = logAppender.getLog(0).toString();
+      assertTrue(logline, logline.matches("^127\\.0\\.0\\.1 - \\[\\d+/\\w+/\\d+:\\d+:\\d+:\\d+ [+-]\\d+\\] \"PUT /HotRodCache/\\[B0x6B6579 HOTROD/2\\.1\" OK \\d+ \\d+ \\d+ ms$"));
+      logAppender.uninstall();
```

Comment:
Remove this in a clean up method? If the assertion fails this logger will never get removed.

Reply:
Right